### PR TITLE
media:Make Decoder visible only to InputDataSource

### PR DESCRIPTION
**Pull request by junmin-kim**
Only InputDataSource can handle Decoder.
Derived datasource call register and unresgister decoder.

### CODE REVIEW

**Comment by junmin-kim** [comment]
@gcjjyy Would you review this PR?

**Comment by seinfra** [comment]
Target : [e01ab168703f80ca739de546e8a75ea791f51845] - Code Rule Check Result:
framework/include/media/InputDataSource.h:82: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
framework/include/media/InputDataSource.h:121: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/InputDataSource.h:121: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:121: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:121: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:142: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/InputDataSource.h:142: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:142: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:142: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxB)
framework/include/media/InputDataSource.h:143: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:143: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:143: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:145: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:145: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:145: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:152: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:152: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:152: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:153: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:153: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:153: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:154: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:154: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:154: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:155: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:155: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:155: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:156: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:156: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/include/media/InputDataSource.h:156: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/include/media/InputDataSource.h:162: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/InputDataSource.h:163: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)

**Comment by seinfra** [comment]
Target : [e01ab168703f80ca739de546e8a75ea791f51845] - Code Rule Check (C++) OK.

File: framework/src/media/InputDataSource.cpp [hunk 1]
```diff
@@ -57,14 +53,34 @@ InputDataSource::~InputDataSource()
...
 
-void InputDataSource::setDecoder(std::shared_ptr<Decoder> decoder)
+void InputDataSource::registerDecoder(audio_type_t audioType, unsigned int channels, unsigned int sampleRate)
 {
+	std::shared_ptr<Decoder> decoder = nullptr;
+	switch (audioType) {
+	case AUDIO_TYPE_MP3:
+	case AUDIO_TYPE_AAC:
+	case AUDIO_TYPE_OPUS:
+		decoder = std::make_shared<Decoder>(channels, sampleRate);
```
Comment (jsdosa) [inline framework/src/media/InputDataSource.cpp]: change decoder to mDecoder

Comment (junmin-kim) [reply framework/src/media/InputDataSource.cpp]: I'll fix it later PR

**Pull request merged**